change/@ni-nimble-blazor-803b27ab-c364-4038-afe2-da2ac9153dbf.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Blazor integration for primary and accent appearance-variant buttons (#663)",
+  "packageName": "@ni/nimble-blazor",
+  "email": "103057880+aidendk@users.noreply.github.com",
+  "dependentChangeType": "patch"
+}
```

Comment:
original from @aidendk :

Regarding Mert's comment about duplication:

This is also something I want to leave open to everyone for their feedback:

Currently there are two ways that I have implemented the styling for the appearance-variants.

**1. Conditionally changing the value of the variable (ex. `:host([appearance-variant='primary'])`)** 

(**Used in most recent commit**)

This implementation gets rid of the need for a MultivaluePropertyStyleSheetBehavior, but requires css variables to be declared multiple times. For example, when the button is in the hover state the outer border should have a color of `borderHoverColor`, as declared in .control:hover towards the top of the css class:

```
    .control:hover {
        border-color: ${borderHoverColor};
        box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
    }
```
Even though this is declared initially, because we are using :host to conditionally change the value of the variable in the `buttonAppearanceVariantStyles`, the `border-color: ${actionRgbPartialColor}` takes over the `${borderHoverColor}` that was initially declared, as it does not have :host.
Becuase of this, the border color has to be changed every time, as shown in the example below.
```
export const buttonAppearanceVariantStyles = css``.withBehaviors(
    appearanceBehavior(
        ButtonAppearance.outline,
        css`
            :host([appearance-variant='primary']) .control {
                border-color: ${actionRgbPartialColor};
            }

            :host([appearance-variant='primary']) .control:hover {
                border-color: ${borderHoverColor};
                background-color: transparent;
            }

           ...
```
<br/>
**2. Using another MultivaluePropertyStyleSheetBehavior for the appearance-variant attribute**

(**Used in [this commit](https://github.com/ni/nimble/blob/6ccd0e42cb63a1f69de83eafb86a2ecb474d323a/packages/nimble-components/src/patterns/button/styles.ts#L266)**)

Using the MultivaluePropertyStyleSheetBehavior does require adding another function to the `appearance.ts` file, but does not require re-applying variables like border-color for previously defined css classes because :host is not used.

```
export const buttonAppearanceVariantStyles = css``.withBehaviors(
    appearanceVariantBehavior(
        ButtonAppearanceVariant.primary,
        css``.withBehaviors(

           ...

            appearanceBehavior(
                ButtonAppearance.outline,
                css`
                    .control {
                        border-color: ${actionRgbPartialColor};
                    }

                    .control:hover {
                        background-color: transparent;
                    }
```
Here, the border-color is still changed for .control because the border color is a different color when the appearance-variant is `primary`. Although, it is not required for the control:hover css class.

Either one works, but each has their own advantages and disadvantages. Let me know what you guys think! 😄